### PR TITLE
Update to conda 23.10 and mamba 1.5.3

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: Miniforge3/construct.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {% set name = os.environ.get("MINIFORGE_NAME", "Miniforge3") %}
-{% set version = os.environ.get("MINIFORGE_VERSION", "23.3.1-0") %}
-{% set conda_libmamba_solver_version = "23.3.0"%}
-{% set mamba_version = "1.4.2"%}
+{% set version = os.environ.get("MINIFORGE_VERSION", "23.7.4-0") %}
+{% set conda_libmamba_solver_version = "23.7.0"%}
```
Comment (hmaarrfk) [inline Miniforge3/construct.yaml:3]:
```suggestion
{% set conda_libmamba_solver_version = "23.9.1"%}
```

**Comment by hmaarrfk** [comment]
@conda-forge/boa I'm curious why the solver can't find a solution for `boa` and `mamba 1.5.1`

https://github.com/conda-forge/boa-feedstock/blob/main/recipe/meta.yaml#L27


```

+ mamba install mamba=1.5.1 'python=3.10.*=*_cpython' boa --yes

Looking for: ['mamba=1.5.1', 'python=3.10[build=*_cpython]', 'boa']

Could not solve for environment specs
The following packages are incompatible
├─ boa is installable with the potential options
│  ├─ boa [0.2.1|0.3.0] would require
│  │  └─ mamba >=0.6.4,<0.15.0a0 , which can be installed;
│  ├─ boa [0.3.1|0.3.2|...|0.4.3] would require
│  │  └─ mamba >=0.7,<0.15.0a0 , which can be installed;
│  ├─ boa 0.5.0 would require
│  │  └─ mamba >=0.11,<0.15.0a0 , which can be installed;
│  ├─ boa 0.6.0 would require
│  │  └─ mamba >=0.14,<0.16 , which can be installed;
│  ├─ boa [0.6.1|0.6.2] would require
│  │  └─ mamba >=0.15,<0.16 , which can be installed;
│  ├─ boa [0.1.0|0.1.2|0.1.3|0.1.4] would require
│  │  └─ mamba 0.5.* , which can be installed;
│  ├─ boa 0.10.0 would require
│  │  └─ mamba >=0.22.0,<0.23 , which can be installed;
│  ├─ boa 0.11.0 would require
│  │  └─ mamba >=0.23.0,<0.24 , which can be installed;
│  ├─ boa 0.11.0 would require
│  │  └─ mamba >=0.23.0,<0.25 , which can be installed;
│  ├─ boa 0.11.0 would require
│  │  └─ mamba >=0.23.0,<0.26 , which can be installed;
│  ├─ boa 0.11.0 would require
│  │  └─ mamba >=0.23.0,<0.28 , which can be installed;
│  ├─ boa 0.12.0 would require
│  │  └─ mamba >=0.27.0,<0.28 , which can be installed;
│  ├─ boa 0.13.0 would require
│  │  └─ mamba >=1.0,<1.1 , which can be installed;
│  ├─ boa 0.2.0 would require
│  │  └─ mamba 0.6.* , which can be installed;
│  ├─ boa [0.7.0|0.7.1] would require
│  │  └─ mamba >=0.16,<0.18 , which can be installed;
│  ├─ boa 0.8.0 would require
│  │  └─ mamba >=0.18,<0.19 , which can be installed;
│  ├─ boa [0.8.1|0.8.2] would require
│  │  └─ mamba >=0.19.1,<0.20 , which can be installed;
│  ├─ boa 0.9.0 would require
│  │  └─ mamba >=0.21.0,<0.22 , which can be installed;
│  ├─ boa 0.14.0 would require
│  │  └─ mamba >=1.0,<1.2 , which can be installed;
│  ├─ boa 0.14.0 would require
│  │  └─ mamba >=1.0,<1.4 , which can be installed;
│  ├─ boa 0.14.0 would require
│  │  └─ mamba >=1.0,<1.5 , which can be installed;
│  └─ boa 0.15.1 would require
│     └─ mamba >=1.0,<=1.4.2 , which can be installed;
└─ mamba 1.5.1**  is not installable because it conflicts with any installable versions previously reported.
```

let me know if you have ideas

**Comment by hmaarrfk** [comment]
hmm, this is why https://github.com/conda-forge/admin-requests/pull/811

**Comment by jezdez** [comment]
@conda-forge-admin, please rerender

File: Miniforge3/construct.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {% set name = os.environ.get("MINIFORGE_NAME", "Miniforge3") %}
-{% set version = os.environ.get("MINIFORGE_VERSION", "23.3.1-0") %}
-{% set conda_libmamba_solver_version = "23.3.0"%}
-{% set mamba_version = "1.4.2"%}
+{% set version = os.environ.get("MINIFORGE_VERSION", "23.10.0-0") %}
+{% set conda_libmamba_solver_version = "23.11.0"%}
```
Comment (jezdez) [inline Miniforge3/construct.yaml:3]:
```suggestion
{% set conda_libmamba_solver_version = "23.11.1"%}
```

**Comment by h-vetinari** [comment]
There's now a boa release that's compatible with mamba 1.5. I've restarted the failed CI jobs.

**Comment by h-vetinari** [comment]
The builds from https://anaconda.org/conda-forge/boa/files are not making it through the CDN at the moment...

**Comment by h-vetinari** [comment]
This looks good now on all platforms except osx, where we get:
```
ImportError: dlopen(/Users/runner/conda/lib/python3.10/site-packages/libmambapy/bindings.cpython-310-darwin.so, 0x0002): Symbol not found: (_solver_ruleinfo2str)
  Referenced from: '/Users/runner/conda/lib/libmamba.2.0.0.dylib'
  Expected in: '/Users/runner/conda/lib/libsolv.1.dylib'
```

CC @jaimergp @wolfv

**Comment by jaimergp** [comment]
Pin for latest libsolv 0.7.26. Let's see if that fixes it.

**Comment by mbargull** [comment]
This fails on macOS during setup after install `constructor`.
It doesn't use latest package versions overall.
We may want to update https://github.com/conda-forge/miniforge/blob/c1cc90a5cef738c96eb791b7c87f8bf3934e4dd3/build_miniforge_osx.sh#L10-L15 to use newer versions for setup.
We apparently already use `setup-miniconda` for Windows -- could we just use that for macOS too instead?

**Comment by mbargull** [comment]
@hmaarrfk, I've changed the macOS builds to use `setup-miniconda` and updated `conda-libmamba-solver=23.11.1` in gh-518 -- feel free to merge or cherry-pick commits from that PR if they look sensible to you.

**Comment by mbargull** [comment]
Also, the decision on if/when we should release this should be come after making a decision on https://github.com/conda-forge/miniforge/issues/511#issuecomment-1816199339 ( https://github.com/conda/conda-libmamba-solver/issues/386 ):
> But some feedstocks are incompatible due to a bug I'm chasing in https://github.com/conda/conda-libmamba-solver/issues/386. These feedstocks need `boa` for `mambabuild`.

**Pull request closed without merging**